btp-document/BMR-Deployment.md

```diff
@@ -0,0 +1,212 @@
+
+BMR2: listen blocks generating from Moonriver Network --> build `Relay Message` --> push this message to ICON-BMC contract.
+
+==> In order to push `Relay Message` to ICON-BMC contract, BMR2 must be registered into ICON-BMC contract. Thus, it requires BMR2's address being generated in ICON Network. Then, `Operator` of Moonriver-BMC contract can manually add this address to handle a connection link (Moonriver --> ICON)
+
+- Confusion point 2: BMR handles a connection from A to B. Similar to the above confusion point, BMR listens block generating on A, but it must query this offet through calling `getBMCLinkStatus` on B
+
+==> To avoid chicken-and-egg problem, `Operator` will choose an `offset` and set it into BMV contract when this contract is deployed. From that point, blocks can be verified and synced. Upon successful verification, BMV updates this `offset` to the latest block height. In addition, current block height on A might be higher than current block height was verified by BMV contract on B. Thus, BMR is required to retrieve the latest `offset` from BMV contract through calling `getBMCLinkStatus` on B
+
+### 1. Build executable files
+
+____
+
+```shell
+# INode is a folder that was created when deploying a local node in ICON Network
+$ CONFIG_DIR=path/to/config/folder/INode
+
+$ cd $CONFIG_DIR
+
+#  Clone BTP Project
+$ git clone https://github.com/icon-project/btp.git
+
+$ cd btp && git checkout refactor-layout
+
+# Output binaries are placed under bin/ directory.
+$ make btpsimple
+```
+
+### 2. Generate keystore and configuration files
+
+____
+
+#### Download CLI tools
+
+```shell
+# Get goloop-cli
+$ go get github.com/icon-project/goloop/cmd/goloop
+
+# Get ethkey-cli
+$ go get github.com/ethereum/go-ethereum/cmd/ethkey
+```
+
+#### Create keystore files
+
+- Generate keystore of BMR from Moonriver --> ICON
+
+```shell
+$ goloop ks gen --out $CONFIG_DIR/icon-bmr.keystore.json --password $YOUR_PASSWORD
+
+# Create a secret file 'icon-bmr.secret' and save $YOUR_PASSWORD into that file
+$ cat > icon-bmr.secret
+# Enter your password
+```
+
+- Generate keystore of BMR from ICON --> Moonriver
+
+```shell
+$ ethkey generate moon-bmr.keystore.json
+# Enter your password
+# Repeat your password
+
+# Create a secret file 'moon-bmr.secret' and save $YOUR_PASSWORD into that file
+$ cat > moon-bmr.secret
+# Enter your password
+```
+
+#### Query `Offset` and a list of `Relays` from BMC
+
+&emsp; This step checks whether **BMRs** have been registered and added into **BMC**. As designed, only registered **BMRs**, that approved and added by `Operator`, are allowed to build `Relay Message` and push it to **BMC** contract
+
+- Get `BMCLinkStatus` (Moonriver --> ICON): please run below commands
+
+```bash
+$ goloop rpc --uri http://goloop:9080/api/v3/icon \
+    call --to $JAVA_SCORE_BMC_ADDRESS \
+    --method getStatus \
+    --param _link= $BTPSIMPLE_SRC_ADDRESS
```

Comment:
Unclear that $BTPSIMPLE_SRC_ADDRESS is `This is btp address of PRA BMC with format btp://0x501/${PRA_BMC_ADDRESS}`